### PR TITLE
[PostgreSQL] SDK upgrade for PostgreSQL flexible server to version 2022-12-01

### PR DESCRIPTION
**Pull request by alanenriqueo**
This release uses GA api version 2022-12-01 for PostgreSQL flexible server.

Features added:
- Data encryption
- User managed identities
- Active Directory authentication
- Geo-Restore
- Read Replicas
- Same Zone HA
- Server backups
- Major version upgrade

---

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by archerzz** [comment]
@alanenriqueo Can you check the build errors? It looks like the codes are out-of-sync: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2129455&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=45862c6e-83c5-515f-73e2-e7009eff9f9b&l=18

**Review by archerzz — COMMENTED** [review]
OK. There are breaking changes in version 2.0.0. And the `<ApiCompatVersion>` is set to `1.0.0`, which broke the build. See [here](https://github.com/Azure/azure-sdk-for-net/blob/807cff414fe3b05747dcafd803ae7fe077cfafa3/sdk/postgresql/Azure.ResourceManager.PostgreSql/src/Azure.ResourceManager.PostgreSql.csproj#L4-L5).

Need to reconsider the api backward compatibility. For details, see [here](https://github.com/Azure/azure-sdk-for-net/blob/807cff414fe3b05747dcafd803ae7fe077cfafa3/CONTRIBUTING.md#api-compatibility-verification).

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/PostgreSqlFlexibleServerDataEncryption.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            PrimaryKeyUri = primaryKeyUri;
+            PrimaryUserAssignedIdentityId = primaryUserAssignedIdentityId;
+            KeyType = keyType;
+        }
+
+        /// <summary> URI for the key for data encryption for primary server. </summary>
+        public Uri PrimaryKeyUri { get; set; }
+        /// <summary> Resource Id for the User assigned identity to be used for data encryption for primary server. </summary>
+        public string PrimaryUserAssignedIdentityId { get; set; }
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/PostgreSqlFlexibleServerDataEncryption.cs:34]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServerActiveDirectoryAdministratorData.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            ObjectId = objectId;
+            TenantId = tenantId;
+        }
+
+        /// <summary> The principal type used to represent the type of Active Directory Administrator. </summary>
+        public PostgreSqlFlexibleServerPrincipalType? PrincipalType { get; set; }
+        /// <summary> Active Directory administrator principal name. </summary>
+        public string PrincipalName { get; set; }
+        /// <summary> The objectId of the Active Directory administrator. </summary>
+        public string ObjectId { get; set; }
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServerActiveDirectoryAdministratorData.cs:45]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

**Comment by archerzz** [comment]
Create an API view: https://apiview.dev/Assemblies/Review/e3c24d89f5b549829fe15b605c8459ce/ff1851a20ea34f2080e2181922cfed4b?diffRevisionId=1f7130b266864ef3bb0874f3f45607ce&doc=False&diffOnly=True

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServerActiveDirectoryAdministratorData.cs (hunk 2 above)
Comment (alanenriqueo) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServerActiveDirectoryAdministratorData.cs:45]: `ObjectId` should be a GUID.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/PostgreSqlFlexibleServerDataEncryption.cs (hunk 1 above)
Comment (alanenriqueo) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/PostgreSqlFlexibleServerDataEncryption.cs:34]: `PrimaryUserAssignedIdentityId` should be a resource identifier.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServerActiveDirectoryAdministratorData.cs (hunk 2 above)
Comment (alanenriqueo) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServerActiveDirectoryAdministratorData.cs]: Fixed.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/PostgreSqlFlexibleServerDataEncryption.cs (hunk 1 above)
Comment (alanenriqueo) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/PostgreSqlFlexibleServerDataEncryption.cs]: Fixed.

**Review by archerzz — APPROVED** [review]

**Pull request merged**